### PR TITLE
Hclang 28 languages for young people

### PR DESCRIPTION
**Pull request by technight**
### Description
Added initial section about programming languages for youngsters and students

### Changelog
 - Added initial section about programming languages for youngsters and students

### Issues solved 
 - HCLANG-28

## FOR REVIEWER [Do not remove]
A small checklist of things to note when you are doing a review. Make sure these things apply before going into master.
 - Did the PR actually fix the issue
 - Comment anything that doesn't make sense or anything you want clarified. Many comments are always ok
 - Make sure this is ready for exams if this is going into master
 - It's always a good idea having another reviewer look at a PR as well, if it's major. get a buddy.

### Regarding code [if applicable]
- Check [coding conventions](https://kotlinlang.org/docs/reference/coding-conventions.html)
- Are public methods documented? 
- Are there testcases for new features / methods?
- Have you tried running the source and tested things that cannot be test-cased?

### Regarding Report [if applicable]
- Does each line have a linechange in the sourcecode?
- Does sentences make sense?
- Did you do spellchecking?
- Citations?

### CODE REVIEW

File: report/2.Analysis/LanguagesForYoungPeople.tex [hunk 1]
```diff
@@ -0,0 +1,63 @@
+\section{Languages for beginners}
+
+There are a wide variety of different tools, games and languages with the desired intend of learning students how to program.
```
Comment (jones1994) [inline report/2.Analysis/LanguagesForYoungPeople.tex]: Line 3:
 * Who/what "desires" the intend? Remove "desired" from sentence.
 * Change "learning" to "teaching"

File: report/2.Analysis/LanguagesForYoungPeople.tex [hunk 2]
```diff
@@ -0,0 +1,63 @@
+\section{Languages for beginners}
+
+There are a wide variety of different tools, games and languages with the desired intend of learning students how to program. 
+These span from very specific game creation tools, to general purpose programming languages. 
+In order to gather some inspiration and knowledge about existing solutions, some of the most common languages for beginners were analysed.
```
Comment (jones1994) [inline report/2.Analysis/LanguagesForYoungPeople.tex]: * State what we as authors are gathering inspiration for, by analyzing the languages below.

File: report/2.Analysis/LanguagesForYoungPeople.tex [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+\subsection{Scratch}
+Scratch is a programming language that lets you program interactive stories, animations and games.
```
Comment (jones1994) [inline report/2.Analysis/LanguagesForYoungPeople.tex]: * Rewrite with the intent to remove "you".

File: report/2.Analysis/LanguagesForYoungPeople.tex [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+There are a wide variety of different tools, games and languages with the desired intend of learning students how to program. 
+These span from very specific game creation tools, to general purpose programming languages. 
+In order to gather some inspiration and knowledge about existing solutions, some of the most common languages for beginners were analysed.
+
+\subsection{Scratch}
+Scratch is a programming language that lets you program interactive stories, animations and games. 
+It is one of the most common programming languages among youngsters and students who are just getting into programming. 
+It is designed for children from the age of 8 to 16, and aims to help children "think creatively, reason systematically, and work collaboratively". 
+In scratch you do not actually have to write any code, since it utilizes drag and drop elements to build and structure your program. Although no exact programming is actually done, scratch helps to bring an understanding of the common programming concepts like variables, loops and conditionals.\cite{ScratchWebsite}
```
Comment (jones1994) [inline report/2.Analysis/LanguagesForYoungPeople.tex]: * Rewrite with the intent to remove "you".

File: report/2.Analysis/LanguagesForYoungPeople.tex [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+Scratch is one of the most popular block building programming languages, but there are a lot of different languages challenging it, for instance google's refinement of scratch which is called blockly.\cite{Blockly}
+
+\subsection{Game maker}
+Game maker is a powerful tool for developing games. It is arguably more complicated than scratch, but also more powerful and still aiming to be easy to use. 
+Quoting their website: 
+
+\textit{"Making games development accessible to everyone means taking away the barriers to getting started. Using our intuitive ‘Drag and Drop’ development environment you can have your game up and running in a matter of minutes without ever having to write any code!".}\cite{GameMaker}
+ 
+Besides also using drag and drop, game maker also created a very simple programming language called "GML".
```
Comment (jones1994) [inline report/2.Analysis/LanguagesForYoungPeople.tex]: * Remove first "also".

File: report/2.Analysis/LanguagesForYoungPeople.tex [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+\subsection{Game maker}
+Game maker is a powerful tool for developing games. It is arguably more complicated than scratch, but also more powerful and still aiming to be easy to use. 
+Quoting their website: 
+
+\textit{"Making games development accessible to everyone means taking away the barriers to getting started. Using our intuitive ‘Drag and Drop’ development environment you can have your game up and running in a matter of minutes without ever having to write any code!".}\cite{GameMaker}
+ 
+Besides also using drag and drop, game maker also created a very simple programming language called "GML". 
+GML resembles the syntax of common languages like C and Java. 
+However, it is a simple language, for instance you should not declare types in it.
```
Comment (jones1994) [inline report/2.Analysis/LanguagesForYoungPeople.tex]: * Rewrite with the intent to remove "you".

File: report/2.Analysis/LanguagesForYoungPeople.tex [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+Game maker is a powerful tool for developing games. It is arguably more complicated than scratch, but also more powerful and still aiming to be easy to use. 
+Quoting their website: 
+
+\textit{"Making games development accessible to everyone means taking away the barriers to getting started. Using our intuitive ‘Drag and Drop’ development environment you can have your game up and running in a matter of minutes without ever having to write any code!".}\cite{GameMaker}
+ 
+Besides also using drag and drop, game maker also created a very simple programming language called "GML". 
+GML resembles the syntax of common languages like C and Java. 
+However, it is a simple language, for instance you should not declare types in it. 
+However, one thing to note is that the type system is indeed static, but types are inferred. 
+As such you may not change the type of a variable at run time.\cite{GML}
```
Comment (jones1994) [inline report/2.Analysis/LanguagesForYoungPeople.tex]: * Rewrite with the intent to remove "you".

File: report/2.Analysis/LanguagesForYoungPeople.tex [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+\end{lstlisting}
+
+The C++ "Hello World" could potentially lead to a lot of questions, and all this just to be able to print to the screen.
+
+Besides the easy and simple syntax python also includes a large collection of libraries to ease the development of different tasks.\cite{PythonLibraries}
+This means you will be using less time reinventing the wheel and thus faster accomplishing the desired end goal.\cite{PythonXKCD}
+
+\subsection{Takeaways}
+Although the above three languages only makes up a small subset of the analysed languages, they seem to be very representative for the general takeaways. 
+Languages intended for beginners will often require you to write little to no code.
```
Comment (jones1994) [inline report/2.Analysis/LanguagesForYoungPeople.tex]: * Rewrite with the intent to remove "you".

File: report/2.Analysis/LanguagesForYoungPeople.tex [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+This means you will be using less time reinventing the wheel and thus faster accomplishing the desired end goal.\cite{PythonXKCD}
+
+\subsection{Takeaways}
+Although the above three languages only makes up a small subset of the analysed languages, they seem to be very representative for the general takeaways. 
+Languages intended for beginners will often require you to write little to no code. 
+This seems to be because they want to learn the students about the general programming concepts, before having to worry about syntax and formalities. 
+Expanding on this, these languages also removes explicit type declaration, making the programmer write less ceremonial code.
+
+Moreover the languages strives to have a simple, easy to learn syntax. 
+Finally the languages generally want to provide a lot of methods and libraries for the user to use, so that he will not have to code these himself.
```
Comment (jones1994) [inline report/2.Analysis/LanguagesForYoungPeople.tex]: * Rewrite with the intent to remove "himself".

File: report/2.Analysis/LanguagesForYoungPeople.tex [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+
+\subsection{Takeaways}
+Although the above three languages only makes up a small subset of the analysed languages, they seem to be very representative for the general takeaways. 
+Languages intended for beginners will often require you to write little to no code. 
+This seems to be because they want to learn the students about the general programming concepts, before having to worry about syntax and formalities. 
+Expanding on this, these languages also removes explicit type declaration, making the programmer write less ceremonial code.
+
+Moreover the languages strives to have a simple, easy to learn syntax. 
+Finally the languages generally want to provide a lot of methods and libraries for the user to use, so that he will not have to code these himself. 
+This enables the user to spend more time telling what he wants done, rather than how he wants it done.
```
Comment (jones1994) [inline report/2.Analysis/LanguagesForYoungPeople.tex]: * Rewrite with the intent to remove "he".

**Comment by technight** [comment]
@jones1994  Tried to comply with your suggestions, would you be so kind to revise it?

**Pull request merged**